### PR TITLE
pkg/envoy/*: add error codes for Envoy RDS and SDS

### PR DESCRIPTION
**Pull request by jaellio**
**Description**:

Adds error codes for errors related to Envoy RDS and SDS.

Part of #2866

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ X ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No
    -   Did you notify the maintainers and provide attribution?

1. Is this a breaking change? No

### CODE REVIEW

File: pkg/errcode/errcode.go [hunk 1]
```diff
@@ -643,13 +682,39 @@ A XDS RBAC policy could not be generated from the specified traffic target
...
 The WASM XDS HttpFilter could not be configured.
+`,
+
+	ErrBuildingRBACPolicyForRoute: `
+A traffic policy rule could not be configured into as a HTTP RBAC route filter.
+The route is built without an HTTP RBAC filter.
+`,
+
+	ErrUnmarshallingSDSCert: `
+A kubernetes ServiceAccount could not be unmarshalled from an SDSCert name.
```
Comment (shashankram) [inline pkg/errcode/errcode.go:699]:
```suggestion
The SDS certificate resource could not be unmarshalled.
The corresponding certificate resource was ignored by the system.
```

File: pkg/errcode/errcode.go [hunk 2]
```diff
@@ -643,13 +682,39 @@ A XDS RBAC policy could not be generated from the specified traffic target
...
+`,
+
+	ErrGettingServiceCertSecret: `
+An XDS Secret containing a new TLS certificate could not be created for the
+certificate request. The certificate request is ignored.
```
Comment (shashankram) [inline pkg/errcode/errcode.go:704]:
```suggestion
An XDS Secret containing a TLS certificate could not be retrieved. 
The corresponding secret request was ignored by the system.
```

File: pkg/errcode/errcode.go [hunk 3]
```diff
@@ -643,13 +682,39 @@ A XDS RBAC policy could not be generated from the specified traffic target
...
+`,
+
+	ErrGettingMeshService: `
+An OSM MeshService could not be configured from a SDSCert name.
```
Comment (shashankram) [inline pkg/errcode/errcode.go:708]:
```suggestion
The SDS secret does not correspond to a MeshService.
```

File: pkg/errcode/errcode.go [hunk 4]
```diff
@@ -643,13 +682,39 @@ A XDS RBAC policy could not be generated from the specified traffic target
...
+`,
+
+	ErrGettingK8sServiceAccount: `
+The SDS resource name could not be parsed and configured as an SDSCert object. The
+certificate request for the invalid SDS resource name is ignored.
```
Comment (shashankram) [inline pkg/errcode/errcode.go:713]:
```suggestion
The SDS secret does not correspond to a ServiceAccount.
```

File: pkg/errcode/errcode.go [hunk 5]
```diff
@@ -643,13 +682,39 @@ A XDS RBAC policy could not be generated from the specified traffic target
...
+`,
+
+	ErrSDSCertMismatch: `
+The identity obtained from the SDSCert request corresponding to the mTLS root
+validation certificate does not match the identity of the proxy.
```
Comment (shashankram) [inline pkg/errcode/errcode.go:718]:
```suggestion
The identity obtained from the SDS certificate request does not match the identity of the proxy.
```

**Review by shashankram — CHANGES_REQUESTED** [review]
Left some comments to clarify the error codes that should be updated both in the description map and the variable comment (if applicable).

File: pkg/errcode/errcode.go (hunk 5 above)
Comment (jaellio) [reply pkg/errcode/errcode.go]: Great suggestions. Thanks for the feedback!

**Review by shashankram — APPROVED** [review]

**Review by allenlsy — APPROVED** [review]

**Pull request merged**